sdk/go/pulumi/rpc.go

```diff
@@ -63,14 +63,68 @@ func mapStructTypes(from, to reflect.Type) func(reflect.Value, int) (reflect.Str
 	}
 }
 
+type urnSet map[URN]struct{}
+
+func (s urnSet) add(v URN) {
+	s[v] = struct{}{}
+}
+
+func (s urnSet) has(v URN) bool {
+	_, ok := s[v]
+	return ok
+}
+
+func (s urnSet) union(other urnSet) {
+	for v := range other {
+		s.add(v)
+	}
+}
+
+func (s urnSet) values() []URN {
+	values := make([]URN, 0, len(s))
+	for v := range s {
+		values = append(values, v)
+	}
+	return values
+}
+
+func addDependency(ctx context.Context, deps urnSet, res Resource) error {
+	if _, custom := res.(CustomResource); !custom {
+		for _, child := range res.getChildren() {
+			if err := addDependency(ctx, deps, child); err != nil {
+				return err
+			}
+		}
+		if !res.isRemoteComponent() {
```

Comment:
I'm a bit confused here. For non-isRemoteComponent, we skip adding `res.URN()` to the `deps`? Why?

Reply:
b/c a local component isn't really part of the dependency graph--instead, it's a stand-in for its descendants. I don't think that there would be any harm in adding local components to the deps, but we've already set a precedent of not doing so in the Node SDK.